account_invoice_report_hide_discounts_by_partner/reports/report_invoice.xml

```diff
@@ -13,5 +13,26 @@
                 <t t-set="display_discount" t-value="o.show_discounts" />
             </t>
         </xpath>
+        <xpath expr="//span[@t-field='line.price_unit']" position="attributes">
+            <attribute name="t-if">display_discount</attribute>
+        </xpath>
+        <xpath expr="//span[@t-field='line.price_unit']" position="after">
+            <t
+                t-set="price_unit_with_disc"
+                t-value="line.price_subtotal/line.quantity"
+                groups="account.group_show_line_subtotals_tax_excluded"
+            />
+            <t
+                t-set="price_unit_with_disc"
+                t-value="line.price_total/line.quantity"
+                groups="account.group_show_line_subtotals_tax_included"
+            />
+            <span
+                t-if="not display_discount"
+                class="text-nowrap"
+                t-out="price_unit_with_disc"
```

Comment:
If user trying to generate the report does not belong to group_show_line_subtotals_tax_excluded or group_show_line_subtotals_tax_included, price_unit_with_disc will not be set.